src/components/Menu/MenuItem.tsx

```diff
@@ -0,0 +1,152 @@
+import { IconExternalLink, IconChevron } from '@/components/Icons';
+import Link from 'next/link';
+import { Platform } from '@/data/platforms';
+
+enum Levels {
+  Category = 1,
+  Subcategory = 2,
+  Page = 3
+}
+
+const pathHelper = (path: string) => {
+  const prefix = 'pages/[platform]/';
+
+  const pathNoPrefix = path.slice(prefix.length).split('.')[0];
+
+  const lastIndex = pathNoPrefix.lastIndexOf('/index');
+
+  if (lastIndex > -1) {
+    return pathNoPrefix.substring(0, lastIndex);
+  } else {
+    return pathNoPrefix;
+  }
+};
+
+type PageNode = {
+  title: string;
+  platforms: string[];
+  children: PageNode[];
+  route: string;
+  isExternal: boolean;
+};
+
+type MenuItemProps = {
+  pageNode: PageNode;
+  parentSetOpen: React.Dispatch<React.SetStateAction<boolean>> | null;
+  level: number;
+  currentPlatform: Platform;
+};
+
+export function MenuItem({
+  pageNode,
+  parentSetOpen,
+  level,
+  currentPlatform
+}: MenuItemProps): ReactElement {
+  const router = useRouter();
+  const [open, setOpen] = useState(false);
+  const toggleDrawer = useCallback(() => {
+    setOpen((prevOpen) => !prevOpen);
+  }, []);
+
+  useEffect(() => {
+    if (current) {
+      if (pageNode.children && pageNode.children.length > 0) {
+        // if we're on a heading that has children, open it
+        setOpen(true);
+      }
+
+      if (parentSetOpen) {
+        // Don't think this scales well with deeply nested menus, what are some better ways to do this?
+        parentSetOpen(true);
+      }
+    }
+  }, []);
+
+  const pathname = `/${currentPlatform}/${pathHelper(pageNode.route)}/`;
+  const current = router.asPath === pathname;
+
+  const currentStyle = current ? 'menu__item--current' : '';
+
+  let categoryStyle = '';
+  switch (level) {
+    case Levels.Category:
+      categoryStyle = 'menu__item--category';
+      break;
+    case Levels.Subcategory:
+      categoryStyle = 'menu__item--subcategory';
+      break;
+    default:
+      categoryStyle = 'menu__item--page';
+      break;
+  }
+
+  if (!pageNode) {
+    return <></>;
+  }
+
+  if (pageNode.isExternal) {
+    return (
+      <li key={pageNode.route}>
+        <AmplifyUILink href={pageNode.route} isExternal={true}>
+          <View
+            className={`menu__item`}
+            onClick={level > Levels.Category ? toggleDrawer : undefined}
+          >
+            <Flex
+              className={`menu__item__inner ${categoryStyle} ${currentStyle}`}
+            >
+              {pageNode.title}
+              <IconExternalLink />
+            </Flex>
+          </View>
+        </AmplifyUILink>
+      </li>
+    );
+  } else if (pageNode.platforms.includes(currentPlatform)) {
+    return (
+      <li key={pageNode.route}>
+        <Link
+          href={{
+            pathname: `/[platform]/${pathHelper(pageNode.route)}`,
+            query: { platform: currentPlatform }
+          }}
+          passHref
+        >
+          <View
```

Comment:
I'm wondering if you might have a better time in the future with link styles and overriding things if you move these classNames to `<Link>` instead and get rid of this `<View>`? Additionally; we have some oddities with behavior since the onClick is on a div inside of a link (we'll need to fix that regardless because of accessibility issues). For instance, if you try keyboard navigating, you cannot expand a link that has children (like Tutorials); it goes to the Tutorials page and does not expand.

I think we have two options (using Tutorials as an example):

- If we want to open Tutorials' children without navigating, it should be a button; and it should probably not navigate you. We address this on the UI site by having a link called "Overview" nested under each toggle-able parent which essentially takes you to that parent page (I wouldn't necessarily say that's ideal but it lets us avoid the weird dual functionality thing). This means you can expand through menu children without having to also navigate.
- If we want to navigate when clicking on Tutorials, I think it should just be a basic link, and when you're on a page it should auto expand it's children. The caveat there is you have to navigate pages to drill down into child pages. Maybe an annoyance if you're on a slow connection and you know where you're going.
- A weird third option is making the chevron/arrow the click-able thing that does the expanding but I think that's kind of weird and not as intuitive to the user.

What do you think? Maybe we can discuss with Scott since this might impact the way our partners expect it to work.

Reply:
Hey @hbuchel thanks so much for the feedback. We can talk about this offline, I think we'll need input from the others

Reply:
@timngyn actually just talked to Scott to ask his feedback (since we both went through this on the UI docs site). I *think* what you currently have going will work if we can figure out the keyboard navigation piece. And that might be helped if your onclick element isn't inside of the link, but on the link instead (comparing to the next.js docs nav). Does that make sense?